From 0eec35cdda2b20991bbd7fa032f9cd12a0de6f5b Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <matthew.gerlach@linux.intel.com>
Date: Fri, 5 Jan 2024 11:05:06 -0800
Subject: [PATCH] fpga: dfl-cxl-cache: fix check of return of vma_lookup()

The function, vma_lookup, return NULL on error, not PTR_ERR.
Fix the error handling accordingly. This problem was reported
by Coverity.

Signed-off-by: Matthew Gerlach <matthew.gerlach@linux.intel.com
---
 drivers/fpga/dfl-cxl-cache.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/fpga/dfl-cxl-cache.c b/drivers/fpga/dfl-cxl-cache.c
index c9d8064ca064..a33da2de81ad 100644
--- a/drivers/fpga/dfl-cxl-cache.c
+++ b/drivers/fpga/dfl-cxl-cache.c
@@ -271,13 +271,11 @@ static long cxl_cache_set_region_read_only(struct dfl_cxl_cache *cxl_cache,
 					   struct dfl_cxl_cache_buffer_region *region)
 {
 	struct vm_area_struct *vma;
-	long ret = 0;
 
 	vma = vma_lookup(current->mm, region->user_addr);
-	if (IS_ERR(vma)) {
-		ret = PTR_ERR(vma);
-		dev_err(cxl_cache->dev, "vma_lookup() failed: %ld\n", ret);
-		return ret;
+	if (!vma) {
+		dev_err(cxl_cache->dev, "vma_lookup() failed\n");
+		return -EINVAL;
 	}
 
 	mmap_write_lock(current->mm);
@@ -293,7 +291,7 @@ static long cxl_cache_set_region_read_only(struct dfl_cxl_cache *cxl_cache,
 	/* Flush all remaining cache entries. */
 	drm_clflush_virt_range(page_address(region->pages[0]), region->length);
 
-	return ret;
+	return 0;
 }
 
 static long cxl_cache_ioctl_numa_buffer_map(struct dfl_cxl_cache *cxl_cache, void __user *arg)
